### PR TITLE
Fix path handling in read function

### PR DESCRIPTION
**Pull request by cheungpat**
Ghost calls the `read()` function with full URL of a file, hence the asset domain prefix should be removed from path first, before fetching the file from GCS.

refs #1

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -68,21 +68,27 @@ class GoogleCloudStorageAdapter extends BaseAdapter {
...
       readStream.on('error', function(err) {
         return reject(err);
       });
       readStream.on('data', function(data) {
         if (contents) {
-          contents = data;
-        } else {
           contents = Buffer.concat([contents, data]);
+        } else {
+          contents = data;
```
Comment (mikenikles) [inline src/index.js:87]: Hi @cheungpat, what is the reason for this change? This changes the if...else block compared to what it currently is in `master`.

File: src/index.js [hunk 2]
```diff
@@ -68,21 +68,27 @@ class GoogleCloudStorageAdapter extends BaseAdapter {
...
         }
       });
       readStream.on('end', function() {
-        return resolve(content);
+        return resolve(contents);
```
Comment (mikenikles) [inline src/index.js:91]: Nice catch! How could this even work prior to this PR... 🤔

File: src/index.js (hunk 1 above)
Comment (cheungpat) [reply src/index.js:87]: The value of `contents` is initially null, so the `else` block is executed first, which initialize the value of `contents`.

If this change is not applied, `Buffer.concat([null, data])` is called, which will cause error.

File: src/index.js (hunk 2 above)
Comment (cheungpat) [reply src/index.js:91]: The `read()` function seems rarely called. The function is used by Ghost to serve favicon, not when serving images.

File: src/index.js (hunk 1 above)
Comment (mikenikles) [reply src/index.js:87]: Ah of course, that makes sense and given your explanation below saying that `read()` is called rarely, it explains why this has not been identified as a bug.

**Comment by mikenikles** [comment]
@cheungpat Thanks so much for your bug fixes and contributions!

**Comment by cheungpat** [comment]
@mikenikles No problem! Thanks for creating this project!

**Comment by mikenikles** [comment]
Released as version 1.4.0.
